[docs] Updated How to Work on Challenge Files to explain how to test individual challenges (#37188)
freeCodeCamp/freeCodeCamp: docs/how-to-work-on-coding-challenges.md

</details>
````

## Testing Challenges

> Before you [create a pull request](how-to-open-a-pull-request.md) for your changes, you need to validate that the changes you have made do not inadvertently cause problems with the challenge.  You do this by performing the following steps:

Reviewer: ```suggestion
Before you [create a pull request](how-to-open-a-pull-request.md) for your changes, you need to validate that the changes you have made do not inadvertently cause problems with the challenge.  To test all challenges run `npm run test:curriculum`. To save time you can limit the tests to one challenge by performing the following steps:
```
That seems like a good way of getting across why you might want to run one test.  Reasonable?

1. In the `.env` file, set the environment variable `TEST_CHALLENGES_FOR_LANGS` to the language of the challange(s) you are needing to test.  The currently accepted values are `english`, `arabic`, `chinese`, `portuguese`, `russian` and `spanish`.

2.  Switch to the `curriculum` directory:

```
  cd curriculum
```

> 3. Run the following for each challenge file for which you have added solution:

Author: @ojeytonwilliams Actually, I think it needs to be:
> 3. Run the following for each challenge file for which you have changed any of the `testStrings` in the `Tests` section or added solutions:

If you agree, change the suggestion, so I can commit it.

> 3. Run the following for each challenge file for which you have added solution:

Reviewer: Is the edit okay? I went with slightly different wording, but it's hardly a deal breaker.

```
npm run test -- -g 'the full English title of the challenge'
```
    

> Once you tested each applicable challenge file using the syntax above and the test results show all tests pass, then you can create a pull request.

Reviewer: ```suggestion
Once you have verified that each challenge you've worked on passes the tests, please create a pull request.
```
This is a touch more concise and little bit more welcoming.  After all, if they've actually read this far, chances are they're motivated to help!
